### PR TITLE
[1password] Add user fields for actor_details, object_details, and aux_details.

### PR DESCRIPTION
**Pull request by Jin-Dhaliwal**
## What does this PR do?

We've added new details fields to the data model that we want to consumed through this integration. An example of the original message would be:

```
                 {
			"uuid": "2CA5O6WU4VOPE225X64BMDJ6M3",
			"timestamp": "2023-06-30T18:02:23.273264Z",
			"actor_uuid": "PPQYFXGORNDIRHFJ43QNOEKZ6A",
			"actor_details": {
				"uuid": "PPQYFXGORNDIRHFJ43QNOEKZ6A",
				"name": "Insights Test",
				"email": "jin.dhaliwal+insightstest@agilebits.com"
			},
			"action": "trvlaway",
			"object_type": "user",
			"object_uuid": "X3TLF2JNIVDKJAXU67SNSHCFBU",
			"object_details": {
				"uuid": "X3TLF2JNIVDKJAXU67SNSHCFBU",
				"name": "Member 7",
				"email": "jin.dhaliwal+m7@agilebits.com"
			},
			"session": {
				"uuid": "RD4MLUQFIZDW5G6MUFFBIYUUBM",
				"login_time": "2023-06-30T18:01:42.300478Z",
				"device_uuid": "gdbhmdzsecieklx5wmgr7qy6oe",
				"ip": "172.19.0.1"
			}
		}
```

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Author's Checklist

- [x] Integration tests pass 
- [x] Integration includes the new details fields in processed event data

## How to test this PR locally

Follow the testing steps outlined here: https://developer.1password.com/docs/events-api/reference/#post-apiv1auditevents

1. Get the elastic stack up and running. 
2. Follow the integration steps to set up the 1Password integration. (You will need a 1password business account)
  - You can create a test account on start.b5dev.com and use Stripe test card `4242424242424242` 
  - You'll need to change the URL in the integration setup to events.b5dev.com
3. On your 1Password account add some members/groups/vaults/items. This will generate events that the integration will capture. 
4. Confirm that the audit events dashboard contains those events and the new details fields are present. 

## Screenshots

Screenshot 2023-07-06 at 12 40 22 PM
Screenshot 2023-07-06 at 12 40 03 PM

### CODE REVIEW

File: packages/1password/data_stream/audit_events/fields/fields.yml [hunk 1]
```diff
@@ -4,18 +4,57 @@
...
+      type: group
+      fields:
+        - name: uuid
+          type: keyword
+          description: The UUID of the actor.
+        - name: name
+          type: keyword
+          description: The name of the actor.
+        - name: email
+          type: text
```
Comment (kcreddy) [inline packages/1password/data_stream/audit_events/fields/fields.yml:20]:
```suggestion
          type: keyword
```

File: packages/1password/data_stream/audit_events/fields/fields.yml [hunk 2]
```diff
@@ -4,18 +4,57 @@
...
+      type: group
+      fields:
+        - name: uuid
+          type: keyword
+          description: The UUID of the object.
+        - name: name
+          type: keyword
+          description: The name of the object.
+        - name: email
+          type: text
```
Comment (kcreddy) [inline packages/1password/data_stream/audit_events/fields/fields.yml:38]:
```suggestion
          type: keyword
```

File: packages/1password/data_stream/audit_events/fields/fields.yml [hunk 3]
```diff
@@ -4,18 +4,57 @@
...
+      type: group
+      fields:
+        - name: uuid
+          type: keyword
+          description: The UUID of the aux resource.
+        - name: name
+          type: keyword
+          description: The name of the aux resource.
+        - name: email
+          type: text
```
Comment (kcreddy) [inline packages/1password/data_stream/audit_events/fields/fields.yml:56]:
```suggestion
          type: keyword
```

**Review by kcreddy — CHANGES_REQUESTED** [review]
Thanks for the contribution! 

Could you also update ingest pipeline to add User ECS mapping? For example, you could add `user.name` field from `actor_details.name` field.

More examples on User ECS field that you could map: https://www.elastic.co/guide/en/ecs/current/ecs-user-usage.html

**Comment by Jin-Dhaliwal** [comment]
Anything left on this one? Would like to get this out asap because we've released the new version of the API and customers will want to start ingesting the new fields.

**Comment by Jin-Dhaliwal** [comment]
@kcreddy is there any other changes needed here or is this ready for merge now?

**Comment by kcreddy** [comment]
Sincere apologies for the delay 🙏🏼 . 

Looks like the CI is failing due to README being not updated. Can you run below command and commit ?

`elastic-package build && elastic-package format && elastic-package lint && elastic-package check && elastic-package build`

**Comment by Jin-Dhaliwal** [comment]
@kcreddy done!

**Comment by kcreddy** [comment]
Okay, now the CI is failing on pipeline tests. Looks like after the `user` ECS fields are added, the pipeline tests are not updated.

Can you run pipeline tests after bringing the stack up?
`eval "$(elastic-package stack shellinit)" && elastic-package test pipeline --generate` and then commit.

**Comment by kcreddy** [comment]
@Jin-Dhaliwal can you please fix the merge conflicts? The CI wouldn't run otherwise. Thanks!

**Comment by Jin-Dhaliwal** [comment]
@kcreddy OK conflicts resolved

**Review by kcreddy — APPROVED** [review]
LGTM 👍🏼

**Comment by kcreddy** [comment]
> How can I fix those failures in CI? I'm not sure what happened. Reading at face value seems like service that the test step needed failed to start?
Is there a way to restart those?

Hey, @Jin-Dhaliwal the issue was with the CI system, your changes are good. 
Once again thanks so much for the contribution 😄

**Pull request merged**